### PR TITLE
AdaptiveSIM: support solution transfers

### PR DESCRIPTION
**Pull request by akva2**
This adds optional support for solution transfer in adaptive simulations. In particular add support for SIMSolverAdap using AdaptiveISolver.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/ASM/LR/ASMu2Dmx.C [hunk 1]
```diff
@@ -1073,7 +1073,8 @@ bool ASMu2Dmx::refine (const LR::RefineData& prm, Vectors& sol)
     for (size_t j = 0; j < m_basis.size(); j++) {
       Vector bVec;
       this->extractNodeVec(solvec, bVec, 0, j+1);
-      LR::extendControlPoints(m_basis[j].get(), bVec, nfx[j]);
+      if (ASMmxBase::Type != ASMmxBase::SUBGRID || j != 0)
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2Dmx.C:1076]: A `size_t` can never be negative so `j > 0`would make more sense here?

File: src/ASM/LR/ASMu2Dmx.C [hunk 2]
```diff
@@ -1110,7 +1111,8 @@ bool ASMu2Dmx::refine (const LR::RefineData& prm, Vectors& sol)
     for (int i = sol.size()-1; i >= 0; i--)
       for (size_t j = 0; j < m_basis.size(); ++j) {
         sol[i].resize(len);
-        LR::contractControlPoints(m_basis[j].get(), sol[i], nfx[j], ofs);
+        if (ASMmxBase::Type != ASMmxBase::SUBGRID || j != 0)
```
Comment (kmokstad) [inline src/ASM/LR/ASMu2Dmx.C:1114]: Same `j > 0`

File: src/Utility/FieldFunctions.h [hunk 3]
```diff
@@ -212,6 +212,10 @@ class FieldFuncStream : public RealFunc, private FieldFuncBase
 class FieldsFuncBase : public FieldFuncHDF5
 {
 protected:
+  //! \brief Construct from vector of fields.
+  //! \param fields Fields to use
+  FieldsFuncBase(const std::vector<Fields*> fields);
```
Comment (kmokstad) [inline src/Utility/FieldFunctions.h:217]: Missing `&` in argument. Now it'll be pass-by-value. Was that intended here? At least it is inconsistent with the other constructors here.

**Review by kmokstad — APPROVED** [review]
Looks fine, except for some small things that needs clarifications.

File: src/ASM/LR/ASMu2Dmx.C (hunk 1 above)
Comment (akva2) [reply src/ASM/LR/ASMu2Dmx.C]: sure.

File: src/ASM/LR/ASMu2Dmx.C (hunk 2 above)
Comment (akva2) [reply src/ASM/LR/ASMu2Dmx.C]: sure

File: src/Utility/FieldFunctions.h (hunk 3 above)
Comment (akva2) [reply src/Utility/FieldFunctions.h]: yeah, that's a whoopsie. should be passed by reference. of course it works fine as-is and copy is cheap but no reason to copy the vector either.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Pull request merged**